components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/internal/IdentityRecoveryServiceComponent.java

```diff
@@ -517,4 +518,20 @@ protected void unsetApplicationManagementService(ApplicationManagementService ap
 
         IdentityRecoveryServiceDataHolder.getInstance().setApplicationManagementService(null);
     }
+
+    @Reference(
+            name = "workflowservice.default",
+            service = org.wso2.carbon.identity.workflow.mgt.WorkflowManagementService.class,
+            cardinality = ReferenceCardinality.MANDATORY,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unsetWorkflowService")
+    protected void setWorkflowService(WorkflowManagementService workflowManagementService) {
+
+        IdentityRecoveryServiceDataHolder.getInstance().setWorkflowManagementService(workflowManagementService);
+    }
+
+    protected void unsetWorkflowService(WorkflowManagementService workflowManagementService) {
+
+        IdentityRecoveryServiceDataHolder.getInstance().setWorkflowManagementService(null);
```

Comment:
This change correctly sets `IdentityRecoveryServiceDataHolder.getInstance().setWorkflowManagementService(null);` when unsetting the workflow service, which is good because it ensures proper cleanup of the service reference and prevents potential memory leaks or stale references.